src/EPR.ProducerContentValidation.Application/Validators/GroupedValidators/ErrorValidators/ConsistentDataSubmissionPeriodsGroupedValidator.cs

```diff
@@ -16,12 +16,17 @@ public ConsistentDataSubmissionPeriodsGroupedValidator(IMapper mapper, IIssueCou
         _issueCountService = issueCountService;
     }
 
-    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows, List<ProducerValidationEventIssueRequest>? warningRows = null)
+    public override async Task ValidateAsync(List<ProducerRow> producerRows, string storeKey, string blobName, List<ProducerValidationEventIssueRequest> errorRows = null, List<ProducerValidationEventIssueRequest>? warningRows = null)
     {
         var remainingErrorCount = await _issueCountService.GetRemainingIssueCapacityAsync(storeKey);
 
-        var firstProducerRow = producerRows.First();
-        var inconsistentDataSubmissionPeriodRow = producerRows.FirstOrDefault(x => x.DataSubmissionPeriod != firstProducerRow.DataSubmissionPeriod);
+        if (producerRows == null || producerRows.Count == 0)
```

Comment:
could you please write 2 unit test on **old code** i.e. 

var firstProducerRow = producerRows.First();
        var inconsistentDataSubmissionPeriodRow = producerRows.FirstOrDefault(x => x.DataSubmissionPeriod != firstProducerRow.DataSubmissionPeriod);

test 1: producerRows is null
Test 2 producerRows is empty 
and then do your code change and re-run the tests and see what happens?

Reply:
Added another test. I had a previous one to check this.